PLTF-11: Доработка PoemsAPI №1

dimatovs opened the pull request:
Реализовать новый эндпоинт (гет запрос) "/poems/authors-and-poems", выполняющий следующую функцию:
-берет всех авторов
-исключает всех авторов, у которых имя начинается с букв В и С
-формирует по каждому автору объект из полей: имя автора, кол-во его стихов (кол-во - рандомно сгенерированное число от 0 до 100)

dimatovs committed:
PLTF-11
добавил эндпоинт getAuthorsAndPoems, который выводит всех авторов и количество их поэм с нумерацией.

SwatAssault commented on poems/src/main/java/com/project/poems/controller/PoemsController.java:
```diff
@@ -28,4 +30,20 @@ public String getAllAuthors() {
         }
         return result.toString();
     }
+   @GetMapping("/authors-and-poems")
```
не хватает пустой строки

dimatovs replied:
Между методами нужно две пустые строки?

SwatAssault commented on poems/src/main/java/com/project/poems/controller/PoemsController.java:
```diff
@@ -28,4 +30,20 @@ public String getAllAuthors() {
+   public String getAuthorsAndPoems() {
+       Random rand = new Random();
+       AllAuthorsDTO authors = poemsService.getAllAuthors();
+       Stream stream = authors.getAuthors().stream();
```
то, что ты добрался до стримов уже хорошо
но работать с ними надо проще
1. не надо создавать отдельно стрим и сохранять его. Все преобразования над коллекцией с помощью стрима не должны возвращать обратно стрим. В твоем случае должны возвращать обратно коллекцию
2. используй вместо charAt() startsWith() - метод стринг

SwatAssault commented on poems/src/main/java/com/project/poems/controller/PoemsController.java:
```diff
@@ -28,4 +30,20 @@ public String getAllAuthors() {
+               .filter(x -> (x.toString().charAt(0) != 'B' & x.toString().charAt(0) != 'C'))
+               .map(x -> "<li>" + x.toString()+ " - " + rand.nextInt(101) + "</li>");
+       StringBuilder result = new StringBuilder();
+       if (authors != null) {
```
зачем здесь эта проверка? что она дает? ты уже выше обращался к authors

SwatAssault reviewed (changes requested):
общее замечание: форматирование

проигнорировано требование ТЗ:
формирует по каждому автору объект из полей: имя автора, кол-во его стихов (кол-во - рандомно сгенерированное число от 0 до 100)

dimatovs committed:
PLTF-11
-Добавлен файл AuthorStats в разделе dto.
-По каждому автору формируется объект, который хранит два поля - имя и кол-во поэм.
-Отформатировал файл PoemsController.

SwatAssault commented on poems/src/main/java/com/project/poems/dto/AuthorStats.java:
```diff
@@ -0,0 +1,28 @@
+
+
+public class AuthorStats {
+    private String authors;
```
почему множественное число?

SwatAssault commented on poems/src/main/java/com/project/poems/dto/AuthorStats.java:
```diff
@@ -0,0 +1,28 @@
+
+public class AuthorStats {
+    private String authors;
+    private int stats;
```
что такое инт статс? что значит число статс?

SwatAssault commented on poems/src/main/java/com/project/poems/controller/PoemsController.java:
```diff
@@ -30,20 +31,26 @@ public String getAllAuthors() {
-           result.append("</ol>");
+       result.append("Авторы: </br> ");
+       result.append("<ol>");
+       for (AuthorStats point : authorsList) {
```
почему point......

SwatAssault commented on poems/src/main/java/com/project/poems/controller/PoemsController.java:
```diff
@@ -30,20 +31,26 @@ public String getAllAuthors() {
-       stream = stream
-               .filter(x -> (x.toString().charAt(0) != 'B' & x.toString().charAt(0) != 'C'))
-               .map(x -> "<li>" + x.toString()+ " - " + rand.nextInt(101) + "</li>");
+       List<AuthorStats> authorsList = authors.getAuthors().stream()
```
переменную уже имеет смысл назвать по-другому, например, statistics

SwatAssault reviewed (changes requested):
вынеси сразу в рамках этой задачи логику в сервис

dimatovs committed:
PLTF-11
переименовал переменные на более "говорящие".

SwatAssault commented on poems/src/main/java/com/project/poems/dto/AuthorStats.java:
```diff
@@ -2,27 +2,27 @@
-    public AuthorStats(String authors, int stats) {
-        this.authors = authors;
-        this.stats = stats;
+    public AuthorStats(String authors, int poemsNumber) {
```
а в кострукторе осталось по старому

dimatovs committed:
PLTF-11
-переименовал переменные на более "говорящие".
-вынес бизнеслогику в класс сервис.
